### PR TITLE
feat: external link templates

### PR DESCRIPTION
**Pull request by Tymek**
## About the changes

Adds support for link templates in projects, allowing reusable URL patterns with placeholders. Includes validation, database changes, updated API schemas, and tests.

### Important files
- `project-service.ts`: Validates and saves link templates.
- `project-store.ts`: Manages database storage of link templates.
- OpenAPI schemas (`project-link-templates-schema.ts`)

## Discussion points
- Validation logic clarity
- Database storage approach

### CODE REVIEW

File: src/lib/openapi/spec/create-feature-naming-pattern-schema.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ export const createFeatureNamingPatternSchema = {
             nullable: true,
             description:
                 'An example of a feature name that matches the pattern. Must itself match the pattern supplied.',
-            example: 'dx.feature1.1-135',
+            example: 'dx.feature.1-135',
```
Comment (Tymek) [inline src/lib/openapi/spec/create-feature-naming-pattern-schema.ts:21]: That was annoying when testing with OpenAPI docs 😅

File: src/lib/openapi/spec/project-link-templates-schema.ts
Comment (Tymek) [inline src/lib/openapi/spec/project-link-templates-schema.ts:1]: I'll reuse it in Enterprise

File: src/lib/types/model.ts [hunk 2]
```diff
@@ -334,6 +335,11 @@ export interface IProjectParam {
     projectId: string;
 }
 
+export interface IProjectLinkTemplate {
+    title?: string;
+    template: string;
```
Comment (kwasniew) [inline src/lib/types/model.ts:340]: urlTemplate to make it more obvious?

File: src/lib/openapi/spec/project-link-templates-schema.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        title: {
+            type: 'string',
+            description: 'The title of the link.',
+            example: 'Code search',
+            nullable: true,
+        },
+        template: {
+            type: 'string',
+            description:
+                'The URL to use as a template. Can contain {{project}} or {{flag}} as placeholders.',
```
Comment (kwasniew) [inline src/lib/openapi/spec/project-link-templates-schema.ts:19]: flag or feature? I think feature is more common at least in our code

File: src/lib/features/project/project-store.ts [hunk 4]
```diff
@@ -265,6 +271,7 @@ class ProjectStore implements IProjectStore {
                     feature_naming_pattern: data.featureNaming?.pattern,
                     feature_naming_example: data.featureNaming?.example,
                     feature_naming_description: data.featureNaming?.description,
+                    link_templates,
```
Comment (kwasniew) [inline src/lib/features/project/project-store.ts:274]: behind a flag

File: src/lib/features/project/project-store.ts [hunk 5]
```diff
@@ -640,6 +647,7 @@ class ProjectStore implements IProjectStore {
                 example: row.feature_naming_example,
                 description: row.feature_naming_description,
             },
+            linkTemplates: row.link_templates,
```
Comment (kwasniew) [inline src/lib/features/project/project-store.ts:650]: behind a flag

File: src/lib/features/project/project-service.ts [hunk 6]
```diff
@@ -304,6 +304,30 @@ export default class ProjectService {
         return featureNaming;
     };
 
+    private validateLinkTemplates = (
+        linkTemplates: IProject['linkTemplates'] | unknown,
```
Comment (kwasniew) [inline src/lib/features/project/project-service.ts:308]: I'd only expect unknown at the boundary of the system before openapi validation. I'm surprised by seeing unknown at the service level

File: src/lib/features/project/project-store.e2e.test.ts [hunk 7]
```diff
@@ -167,3 +167,62 @@ test('should add environment to project', async () => {
 
     expect(envs).toHaveLength(1);
 });
+
+test('should update project enterprise settings', async () => {
```
Comment (kwasniew) [inline src/lib/features/project/project-store.e2e.test.ts:171]: didn't we have existing tests in this repo or the enterprise repo?

File: src/lib/openapi/spec/project-link-templates-schema.ts (hunk 3 above)
Comment (Tymek) [reply src/lib/openapi/spec/project-link-templates-schema.ts:19]: yes, because we had 'feature' as in our models in the beginning, and UI is moving in this direction. should backend keep old convention?

File: src/lib/features/project/project-service.ts [hunk 8]
```diff
@@ -304,6 +304,30 @@ export default class ProjectService {
...
+    ): IProject['linkTemplates'] => {
+        if (!linkTemplates || !Array.isArray(linkTemplates)) {
+            throw new BadDataError(
+                'The link templates you provided were invalid.',
+            );
+        }
+
+        for (const linkTemplate of linkTemplates) {
```
Comment (kwasniew) [inline src/lib/features/project/project-service.ts:316]: where do we test validation corner cases?

Comment (kwasniew) [reply src/lib/features/project/project-service.ts:316]: if it's difficult to test via service it can be extracted into validation function and tested in isolation

File: src/lib/features/project/project-store.e2e.test.ts (hunk 7 above)
Comment (Tymek) [reply src/lib/features/project/project-store.e2e.test.ts:171]: No. We only have scheme test in Enterprise
